packet_analysis/sql/dao/dns.py

```diff
@@ -0,0 +1,46 @@
+from scapy.all import Packet
+from packet_analysis.sql.dao.sqlObject import sqlObject
+from packet_analysis.sql.dao.udp import UDP, table_sql as udp_table_sql
+from packet_analysis.sql.dao.tcp import TCP, table_sql as udp_table_sql
+
+def table_sql() -> str:
+    """
+    Constructs the necessary parameters for the table building as a string
+    :return: str
+    """
+    return udp_table_sql() + \
+           """,
+              is_tcp   BOOLEAN     NOT NULL,
+              qname    STRING (40) NOT NULL,
+              qtype    STRING (40) NOT NULL,
+              qclass   STRING (40) NOT NULL"""
+
+class DNS(sqlObject):
```

Comment:
I thought DNS used UDP by default? Why do we need to check whether there is a TCP layer in the packet?

Reply:
DNS uses both, generally uses UDP but TCP is possible (Usually used for longer messages).